cluster/event_map.go

```diff
@@ -0,0 +1,71 @@
+package cluster
+
+import (
+	"errors"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// ClusterEventHandlers is a map of EventHandler
+type ClusterEventHandlers struct {
+	sync.RWMutex
+	eventHandlers map[EventHandler]struct{}
+}
+
+// NewClusterEventHandlers returns an EventHandlers
+func NewClusterEventHandlers() *ClusterEventHandlers {
+	return &ClusterEventHandlers{
+		eventHandlers: make(map[EventHandler]struct{}),
+	}
+}
+
+// HandleAll callbacks for the events
+func (eh *ClusterEventHandlers) HandleAll(e *Event) {
```

Comment:
The benefit to the rename is that we don't have to implement a `Cluster.Handle` method at all, if we embed `Cluster.ClusterEventHandlers`